compiler/tools/suggest.nim

```diff
@@ -518,20 +508,13 @@ proc suggestSym*(g: ModuleGraph; info: TLineInfo; s: PSym; usageSym: var PSym; i
   ## misnamed: should be 'symDeclared'
   let conf = g.config
   when defined(nimsuggest):
-    if conf.suggestVersion == 0:
-      if s.allUsages.len == 0:
-        s.allUsages = @[info]
-      else:
-        s.addNoDup(info)
+    if s.allUsages.len == 0:
+      s.allUsages = @[info]
+    else:
+      s.addNoDup(info)
 
-    if conf.ideCmd == ideUse:
-      findUsages(g, info, s, usageSym)
-    elif conf.ideCmd == ideDef:
+    if conf.ideCmd == ideDef:
       findDefinition(g, info, s, usageSym)
-    elif conf.ideCmd == ideDus and s != nil:
-      if isTracked(info, conf.m.trackPos, s.name.s.len):
-        suggestResult(conf, symToSuggest(g, s, isLocal=false, ideDef, info, 100, PrefixMatch.None, false, 0))
-      findUsages(g, info, s, usageSym)
```

Comment:
Why did you remove this as a whole? Only the `findUsage` call is obsolete, as far as I can tell.

Reply:
`dus` is alias for `use`
```nim
if conf.ideCmd in {ideUse, ideDus}:
    let u = if conf.suggestVersion != 1: graph.symFromInfo(conf.m.trackPos) else: graph.usageSym
    if u != nil:
      listUsages(graph, u)
```

Reply:
While `dus` is not an alias for `use`, they indeed perform effectively the same thing in the end. Since it is related, I would suggest removing `ideDus` as part of this PR.

It's not immediately obvious from the code, so please also document in the PR message why `dus` is obsolete and subsumed by `use`.

Reply:
I suggest preserve `ideDus`, it covered by `ideUse` just because `ideUse` also list definition

Reply:
I'm aware, but with the removal of the lines I asked about, you removed the only thing that made `ideDus` slightly different compared to `ideUse`. Removing `ideDus` (but not the `dus` command) would make no difference now.

Reply:
"Removing ideDus (but not the dus command) " I don't understand, there's no `dus` command

Reply:
There is, see the parsing logic within `nimsuggest.execCmd`.

Reply:
I prefer remain these, `ideUse` will be conditional list definition in future.

Reply:
Well, okay, but then what is the point of removing `ideDus` handling?

Also, do you have your plans regarding `nimsuggest` written down somewhere public? If not, please do so.

Reply:
remove `ideDus` handling because covered by `listUsages`, where the place to write down the plan?

Reply:
Not entirely, the extra `suggestResult` call is not covered by `listUsages`. Sure, it results in a duplicate 'def' being reported, as `listUsages` already reports definitions, but still.

For writing down the plan, I think a GitHub discussion makes sense.

Reply:
okay, I revert `ideDus` handle, it's fine without this change in this PR.